src/PostgREST/Auth.hs

```diff
@@ -54,33 +55,48 @@ import Protolude
 
 -- | Receives the JWT secret and audience (from config) and a JWT and returns a
 -- JSON object of JWT claims.
-parseToken :: AppConfig -> ByteString -> UTCTime -> ExceptT Error IO JSON.Value
-parseToken _ "" _ = return JSON.emptyObject
-parseToken AppConfig{..} token time = do
-  secret <- liftEither . maybeToRight (JwtErr JwtTokenMissing) $ configJWKS
-  eitherContent <- liftIO $ JWT.decode (JWT.keys secret) Nothing token
+parseToken :: AppConfig -> Maybe ByteString -> UTCTime -> ExceptT Error IO JSON.Value
+parseToken _ Nothing _ = return JSON.emptyObject
+parseToken _ (Just "") _ = throwE . JwtErr $ JwtDecodeError "Empty JWT is sent in Authorization header"
+parseToken AppConfig{..} (Just tkn) time = do
+  -- validate if the token has 3 period separated parts and min of 32 chars
+  min32CharToken <- liftEither $ hasMin32Chars tkn
+  tknWithThreeParts <- liftEither $ hasThreeParts min32CharToken
+  secret <- liftEither . maybeToRight (JwtErr JwtSecretMissing) $ configJWKS
+  eitherContent <- liftIO $ JWT.decode (JWT.keys secret) Nothing tknWithThreeParts
   content <- liftEither . mapLeft (JwtErr . jwtDecodeError) $ eitherContent
   liftEither $ mapLeft JwtErr $ verifyClaims content
   where
-      -- TODO: Improve errors, those were just taken as-is from hs-jose to avoid
-      -- breaking changes.
+      hasThreeParts :: ByteString -> Either Error ByteString
+      hasThreeParts token = case length $ BS.split (BS.c2w '.') token of
+        3 -> Right token
+        n -> Left $ JwtErr $ JwtDecodeError ("Expected 3 parts in JWT; got " <> show n)
+      hasMin32Chars :: ByteString -> Either Error ByteString
+      hasMin32Chars token
+          | BS.length token >= 32 = Right token
+          | otherwise = Left $ JwtErr $ JwtDecodeError "JWT must be at least 32 characters"
```

Comment:
Wasn't the `jwt-secret` the one that needed to be at least 32 chars? Not the JWT that's sent per request?

Reply:
Hmm, now that I think of it, it is unnecessary. I just thought it would be a little nice to give this error, I'll remove it.